main.js

```diff
@@ -0,0 +1,96 @@
+const { InstanceBase, Regex, runEntrypoint, InstanceStatus, TCPHelper } = require('@companion-module/base')
+const UpgradeScripts = require('./upgrades')
+const UpdateActions = require('./actions')
+const UpdateFeedbacks = require('./feedbacks')
+const UpdateVariableDefinitions = require('./variables')
+const presets = require('./presets')
+
+class ModuleInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
+		Object.assign(this, { ...presets })
+	}
+
+	async init(config) {
+		this.config = config
+
+		this.updateStatus(InstanceStatus.Ok)
+
+		if (this.socket) {
+			this.socket.destroy()
+			delete this.socket
+		}
+
+		this.updateStatus(InstanceStatus.Connecting)
+
+		if (this.config.host) {
+			this.socket = new TCPHelper(this.config.host, 23)
+
+			this.socket.on('status_change', (status, message) => {
+				this.updateStatus(status, message)
+			})
+
+			this.socket.on('error', (err) => {
+				this.updateStatus(InstanceStatus.ConnectionFailure, err.message)
+				this.log('error', 'Network error: ' + err.message)
+			})
+
+			this.socket.on('data', (data) => {
+				if (this.config.saveresponse) {
+					let dataResponse = data
+
+					if (this.config.convertresponse == 'string') {
```

Comment:
you don't have a config field convertresponse, this expression will always be false

Reply:
removed